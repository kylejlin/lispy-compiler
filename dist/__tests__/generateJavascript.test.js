'use strict';

var _createAST = require('../createAST');

var _createAST2 = _interopRequireDefault(_createAST);

var _tokenize = require('../tokenize');

var _tokenize2 = _interopRequireDefault(_tokenize);

var _generateJavascript = require('../generateJavascript');

var _generateJavascript2 = _interopRequireDefault(_generateJavascript);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var code = '\n(pickRandom "foo" 9 (sum 1 2 3))\n';

var tokens = (0, _tokenize2.default)(code);
var ast = (0, _createAST2.default)(tokens);
var js = (0, _generateJavascript2.default)(ast);

test('it generates the correct js', function () {
  expect(js).toMatchSnapshot();
});