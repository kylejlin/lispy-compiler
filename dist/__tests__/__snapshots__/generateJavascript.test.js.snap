// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it generates the correct js 1`] = `
"/** Generated by lispy-compiler **/

pickRandom(
  'foo',
  9,
  sum(
    1,
    2,
    3,
  )
)
"
`;
